OrbitQt/AccessibilityTests.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <QAccessibleWidget>
+#include <QApplication>
+#include <list>
+
+#include "AccessibilityAdapter.h"
+#include "AccessibilityInterfaceMock.h"
+#include "orbitglwidget.h"
+
+namespace orbit_qt {
+
+using orbit_gl::GlAccessibleInterface, orbit_gl::A11yRole, orbit_gl::A11yRect, orbit_gl::A11yState;
+using orbit_gl::TestA11yImpl;
+
+TEST(Accessibility, CreationAndManagement) {
+  TestA11yImpl* obj = new TestA11yImpl(nullptr);
+  QAccessibleInterface* a1 = A11yAdapter::GetOrCreateAdapter(obj);
+  EXPECT_TRUE(a1->isValid());
+
+  QAccessibleInterface* a2 = A11yAdapter::GetOrCreateAdapter(obj);
+  EXPECT_TRUE(a2->isValid());
+  EXPECT_EQ(a1, a2);
+
+  delete obj;
+  EXPECT_EQ(A11yAdapter::RegisteredAdapterCount(), 0);
+}
+
+TEST(Accessibility, ExternalWidget) {
+  QObject object;
+
+  // This should usually happen automatically - OrbitGlWidgetAccessible will register
+  // and unregister on construction and deletion. This can currently not be tested though because we
+  // cannot create a OrbitGlWidgetAccessible without an OrbitGlWidget, and this has too many
+  // dependencies to be instantiated in a unit test at this point
+  TestA11yImpl* obj = new TestA11yImpl(nullptr);
```

Comment:
`unique_ptr` or `optional`.

Reply:
Agree here since I'm testing the explicit de-registration, changed.